modules/node-shared/src/main/scala/io/constellationnetwork/node/shared/infrastructure/snapshot/GlobalSnapshotAcceptanceManager.scala

```diff
@@ -1168,34 +1168,36 @@ object GlobalSnapshotAcceptanceManager {
       acceptedSpendActions: Map[Address, List[SpendAction]],
       currentGlobalOrdinal: SnapshotOrdinal,
       currentGlobalEpochProgress: EpochProgress
-    ): SortedMap[Address, MetagraphSyncDataInfo] =
-      lastSnapshotContext.metagraphSyncData.map { existingData =>
-        val updatedFromSnapshots = updateFromCurrencySnapshots(
-          existingData,
-          currencySnapshots,
-          globalSnapshotsProcessed,
-          currentGlobalOrdinal,
-          currentGlobalEpochProgress
-        )
+    ): F[SortedMap[Address, MetagraphSyncDataInfo]] =
+      lastSnapshotContext.metagraphSyncData match {
+        case None => SortedMap.empty[Address, MetagraphSyncDataInfo].pure[F]
+        case Some(existingData) =>
+          for {
+            updatedFromSnapshots <- updateFromCurrencySnapshots(
+              existingData,
+              currencySnapshots,
+              globalSnapshotsProcessed,
+              currentGlobalOrdinal,
+              currentGlobalEpochProgress
+            )
 
-        val updatedFromSpendActions = updateFromSpendActions(
-          updatedFromSnapshots,
-          acceptedSpendActions,
-          currentGlobalOrdinal
-        )
+            updatedFromSpendActions = updateFromSpendActions(
+              updatedFromSnapshots,
+              acceptedSpendActions,
+              currentGlobalOrdinal
+            )
 
-        existingData ++ updatedFromSpendActions
+          } yield existingData ++ updatedFromSpendActions
       }
```

Comment:
we should stick with the `.map` here. Excessive pattern matching will be detrimental. The logic seems nearly identical.